pythontest.py

```diff
@@ -42,71 +42,88 @@ def reading_file(file):
-    #         price = origin['price'][origin.id == id_value]
-    #         # Set value of destination file price to this price
-    #         destination['price'][destination.id == id_value] = price
-
+            sys.exit('The file does not have the right header columns. Please provide this column: ' + column)
 
 
 def file_copy(origin, destination):
     # iterate over ids in the destination file
+    # Set-up counter for matches
+    i = 0
+    # Set-up counter for where the price was already correct
+    z = 0
     for index in range(len(destination)):
         # find all items in origin that match that item id
         filterId = origin['id'] == destination.loc[index, 'id']
         matchesId = origin.loc[filterId]
+        # create counter for changed id lines
+
         # if there are no matched we do nothing, otherwise we copy over the price
         if len(matchesId) > 0:
             destination.loc[index, 'price'] = matchesId.iloc[0]['price']
-        # find all items in origin that match the name column
-        # TODO: Convert all names to lower or upper case
-        filterName = origin['name'] == destination.loc[index, 'name']
+            i += 1
+        else:
+            # find all items in origin that match the name column
+            filterName = origin['name'].str.upper() == destination.loc[index, 'name'].upper()
+
+            matchesName = origin.loc[filterName]
+            if len(matchesName) > 0:
+                if destination.loc[index, 'price'] == matchesName.iloc[0]['price']:
+                    z += 1
+                else:
+                    destination.loc[index, 'price'] = matchesName.iloc[0]['price']
+                    i += 1
+
+    # Report for amount of rows in origin
+    lines_origin = len(origin.axes[0])
+    print('Number of items in the origin file ', lines_origin)
+
+    # Report for amount of rows in destination
+    lines_destination = len(destination.axes[0])
+    print('Number of items in the destination file ', lines_destination)
+
+    # Report for changed lines
+    print('We have changed ', i, ' Items')
+    # Report for items that could not be matched
+    print('We were not able to match ', (lines_origin - ( i + z )) ,' item(s)')
+    # Report for lines where the price was already correct
+    print('We could not match ', z, ' Item(s)')
+
+
+def writing_file(file, dataframe, mode = 'yes'):
+    file_extension = getExtension(file)
+    if file_extension == '.xlsx' or file_extension == '.xls':
+        # Reading excel and storing it into dataframe
+        # TODO: Make it pretty
+        if mode == 'no':
+            return dataframe.to_excel(file + '_edited.xlsx', index=False)
+        if mode == 'yes':
+            return dataframe.to_excel(file, index=False)
+    else:
+        # Reading csv and storing it into dataframe
+        if mode == 'no':
+            return dataframe.to_csv(file + '_edited.xlsx', engine='xlsxwriter')
```

Comment:
If the original file was csv we probably want the output to be csv as well. But nice that you found how to do this.